swarm/src/handler/select.rs

```diff
@@ -271,6 +283,24 @@ where
         Poll::Pending
     }
 
+    fn poll_close(&mut self, cx: &mut Context<'_>) -> Poll<Option<Self::ToBehaviour>> {
+        loop {
+            match self.closing_state {
```

Comment:
Why do we need the closing state? Can't we just proceed to closing `proto2` after the first one returned `Poll::Ready(None)`?